File: 28.js
/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 */
var strStr = function (haystack, needle) {
    const regex = new RegExp(needle, 'g');
    const match = haystack.search(regex)

    if (match !== undefined) {
        return match
    } else {
        return -1
    }
};

console.assert(strStr("sadbutsad", "sad") === 0)

File: 28.rs
pub fn str_str(haystack: String, needle: String) -> i32 {
    match haystack.match_indices(&needle).next() {
        Some((idx, _)) => idx as _,
        None => -1,
    }
}

fn main() {
    assert_eq!(str_str("sadbutsad".into(), "sad".into()), 0);
    assert_eq!(str_str("hello".into(), "ll".into()), 2)
}
